grobian@gentoo.org:
The provided libresolv library is faulty for 64-bits, or this call is
doing something wrong.  It is highly undocumented, so it feels like
messing around with some pointers is not going to work.  Doing this in
64-bits mode, at least results in a nice Bus Error.

--- shell_cmds-118/w/w.c
+++ shell_cmds-118/w/w.c
@@ -262,7 +262,7 @@
 	argc -= optind;
 	argv += optind;
 
-#if !TARGET_OS_EMBEDDED
+#if !defined(TARGET_OS_EMBEDDED) && !defined(__LP64__)
 	if (!(_res.options & RES_INIT))
 		res_init();
 	_res.retrans = 2;	/* resolver timeout to 2 seconds per try */
